src/pixel-ping.coffee

```diff
@@ -45,15 +45,20 @@ merge = (new_store) ->
 flush = ->
   log store
   return unless config.endpoint
+  on_error = (message) ->
+    if !config.discard
+      merge(old_store)
+    console.log message
   data = serialize()
   old_store = reset()
   endReqOpts['headers']['Content-Length'] = data.length
-  request = http.request endReqOpts, (response) ->
-    console.info '--- flushed ---'
+  request = http.request endReqOpts, (res) ->
+    if res.statusCode <= 299 and res.statusCode >= 200
```

Comment:
Is everything in `200-299` valid?

I would expect you to explicitly want to check against ` == 200`?

https://www.iana.org/assignments/http-status-codes/http-status-codes.xhtml

Reply:
Everything in the 200 range is "Success". 200, 201, 204 are all very common success response codes (we produce them via our API). I don't see why we'd want to treat anything in the 200 range as failure.